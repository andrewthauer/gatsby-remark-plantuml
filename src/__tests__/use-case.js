// See http://plantuml.com/use-case-diagram

const testRemarkPlugin = require(`../../test/test-remark-plugin`)

describe(`Use Case`, () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it(`Usecases`, async () => {
    const code = `
\`\`\`plantuml
@startuml

(First usecase)
(Another usecase) as (UC2)
usecase UC3
usecase (Last\\nusecase) as UC4

@enduml
\`\`\`
`

    await testRemarkPlugin.testPlugin({
      code,
    })
  })

  it(`Actors`, async () => {
    const code = `
\`\`\`plantuml
@startuml

:First Actor:
:Another\\nactor: as Men2
actor Men3
actor :Last actor: as Men4

@enduml
\`\`\`
`

    await testRemarkPlugin.testPlugin({
      code,
    })
  })
})
