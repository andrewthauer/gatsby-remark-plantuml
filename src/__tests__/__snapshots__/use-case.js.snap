// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use Case Actors 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "lang": undefined,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 11,
          "offset": 113,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "html",
      "value": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" contentScriptType=\\"application/ecmascript\\" contentStyleType=\\"text/css\\" height=\\"265px\\" preserveAspectRatio=\\"none\\" style=\\"width:168px;height:265px;\\" version=\\"1.1\\" viewBox=\\"0 0 168 265\\" width=\\"168px\\" zoomAndPan=\\"magnify\\"><defs><filter height=\\"300%\\" id=\\"fvdb3ugr927tt\\" width=\\"300%\\" x=\\"-1\\" y=\\"-1\\"><feGaussianBlur result=\\"blurOut\\" stdDeviation=\\"2.0\\"/><feColorMatrix in=\\"blurOut\\" result=\\"blurOut2\\" type=\\"matrix\\" values=\\"0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 .4 0\\"/><feOffset dx=\\"4.0\\" dy=\\"4.0\\" in=\\"blurOut2\\" result=\\"blurOut3\\"/><feBlend in=\\"SourceGraphic\\" in2=\\"blurOut3\\" mode=\\"normal\\"/></filter></defs><g><!--entity First Actor--><ellipse cx=\\"38.5\\" cy=\\"26.5\\" fill=\\"#FEFECE\\" filter=\\"url(#fvdb3ugr927tt)\\" rx=\\"8\\" ry=\\"8\\" style=\\"stroke: #A80036; stroke-width: 2.0;\\"/><path d=\\"M38.5,34.5 L38.5,61.5 M25.5,42.5 L51.5,42.5 M38.5,61.5 L25.5,76.5 M38.5,61.5 L51.5,76.5 \\" fill=\\"none\\" filter=\\"url(#fvdb3ugr927tt)\\" style=\\"stroke: #A80036; stroke-width: 2.0;\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"65\\" x=\\"6\\" y=\\"98.0332\\">First Actor</text><!--entity Men2--><ellipse cx=\\"131.5\\" cy=\\"18\\" fill=\\"#FEFECE\\" filter=\\"url(#fvdb3ugr927tt)\\" rx=\\"8\\" ry=\\"8\\" style=\\"stroke: #A80036; stroke-width: 2.0;\\"/><path d=\\"M131.5,26 L131.5,53 M118.5,34 L144.5,34 M131.5,53 L118.5,68 M131.5,53 L144.5,68 \\" fill=\\"none\\" filter=\\"url(#fvdb3ugr927tt)\\" style=\\"stroke: #A80036; stroke-width: 2.0;\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"50\\" x=\\"106.5\\" y=\\"89.5332\\">Another</text><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"32\\" x=\\"106.5\\" y=\\"107.1426\\">actor</text><!--entity Men3--><ellipse cx=\\"38.5\\" cy=\\"180\\" fill=\\"#FEFECE\\" filter=\\"url(#fvdb3ugr927tt)\\" rx=\\"8\\" ry=\\"8\\" style=\\"stroke: #A80036; stroke-width: 2.0;\\"/><path d=\\"M38.5,188 L38.5,215 M25.5,196 L51.5,196 M38.5,215 L25.5,230 M38.5,215 L51.5,230 \\" fill=\\"none\\" filter=\\"url(#fvdb3ugr927tt)\\" style=\\"stroke: #A80036; stroke-width: 2.0;\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"35\\" x=\\"21\\" y=\\"251.5332\\">Men3</text><!--entity Men4--><ellipse cx=\\"122.5\\" cy=\\"180\\" fill=\\"#FEFECE\\" filter=\\"url(#fvdb3ugr927tt)\\" rx=\\"8\\" ry=\\"8\\" style=\\"stroke: #A80036; stroke-width: 2.0;\\"/><path d=\\"M122.5,188 L122.5,215 M109.5,196 L135.5,196 M122.5,215 L109.5,230 M122.5,215 L135.5,230 \\" fill=\\"none\\" filter=\\"url(#fvdb3ugr927tt)\\" style=\\"stroke: #A80036; stroke-width: 2.0;\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"63\\" x=\\"91\\" y=\\"251.5332\\">Last actor</text><!--link First Actor to Men2--><!--link First Actor to Men3--><!--link Men3 to Men4--><!--
@startuml

:First Actor:
:Another\\\\nactor: as Men2
actor Men3
actor :Last actor: as Men4

@enduml


--></g></svg>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 12,
      "offset": 114,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Use Case Usecases 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "lang": undefined,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 11,
          "offset": 122,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "html",
      "value": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" contentScriptType=\\"application/ecmascript\\" contentStyleType=\\"text/css\\" height=\\"164px\\" preserveAspectRatio=\\"none\\" style=\\"width:323px;height:164px;\\" version=\\"1.1\\" viewBox=\\"0 0 323 164\\" width=\\"323px\\" zoomAndPan=\\"magnify\\"><defs><filter height=\\"300%\\" id=\\"fjzl091g0j889\\" width=\\"300%\\" x=\\"-1\\" y=\\"-1\\"><feGaussianBlur result=\\"blurOut\\" stdDeviation=\\"2.0\\"/><feColorMatrix in=\\"blurOut\\" result=\\"blurOut2\\" type=\\"matrix\\" values=\\"0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 .4 0\\"/><feOffset dx=\\"4.0\\" dy=\\"4.0\\" in=\\"blurOut2\\" result=\\"blurOut3\\"/><feBlend in=\\"SourceGraphic\\" in2=\\"blurOut3\\" mode=\\"normal\\"/></filter></defs><g><ellipse cx=\\"69.1041\\" cy=\\"24.9517\\" fill=\\"#FEFECE\\" filter=\\"url(#fjzl091g0j889)\\" rx=\\"63.1041\\" ry=\\"15.4517\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"85\\" x=\\"26.6041\\" y=\\"30.2564\\">First usecase</text><ellipse cx=\\"239.7843\\" cy=\\"24.8569\\" fill=\\"#FEFECE\\" filter=\\"url(#fjzl091g0j889)\\" rx=\\"72.2843\\" ry=\\"16.8569\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"107\\" x=\\"186.2843\\" y=\\"30.1615\\">Another usecase</text><ellipse cx=\\"69.0919\\" cy=\\"127.4517\\" fill=\\"#FEFECE\\" filter=\\"url(#fjzl091g0j889)\\" rx=\\"22.0919\\" ry=\\"15.4517\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"27\\" x=\\"55.5919\\" y=\\"132.7564\\">UC3</text><ellipse cx=\\"163.1843\\" cy=\\"127.7156\\" fill=\\"#FEFECE\\" filter=\\"url(#fjzl091g0j889)\\" rx=\\"37.1843\\" ry=\\"25.7156\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"27\\" x=\\"149.6843\\" y=\\"120.9558\\">Last</text><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"53\\" x=\\"136.6843\\" y=\\"138.5652\\">usecase</text><!--link First usecase to UC2--><!--link First usecase to UC3--><!--link UC3 to UC4--><!--
@startuml

(First usecase)
(Another usecase) as (UC2)
usecase UC3
usecase (Last\\\\nusecase) as UC4

@enduml


--></g></svg>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 12,
      "offset": 123,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
